### PR TITLE
[Core] Geometry based connectivity preserve modeler

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
This extends current capabilities of the `ConnectivityPreserveModeler` to consider geometries. Also note that if neither element type nor condition are provided, now we apply the `ConnectivityPreserveModeler` with nodes and conditions (before we used to throw an error). This makes possible to leverage this model in the geometry-based I/O (e.g., CHT or buoyancy).

### CODE REVIEW

File: kratos/modeler/connectivity_preserve_modeler.h [hunk 1]
```diff
@@ -131,6 +135,22 @@ class KRATOS_API(KRATOS_CORE) ConnectivityPreserveModeler : public Modeler
         const Condition& rReferenceCondition
     );
 
+    /// Generate a copy of rOriginModelPart in rDestinationModelPart.
+    /** This function fills rDestinationModelPart using data obtained from
```
Comment (loumalouomega) [inline kratos/modeler/connectivity_preserve_modeler.h]:
```suggestion
    /**
     * @briefGenerate a copy of rOriginModelPart in rDestinationModelPart.
     * @details This function fills rDestinationModelPart using data obtained from
```

**Review by loumalouomega — DISMISSED** [review]

File: kratos/modeler/connectivity_preserve_modeler.h (hunk 1 above)
Comment (matekelemen) [reply kratos/modeler/connectivity_preserve_modeler.h]: you're missing a space between *brief* and *Generate*

Comment (rubenzorrilla) [reply kratos/modeler/connectivity_preserve_modeler.h]: Done. Thanks for noting it.

**Comment by rubenzorrilla** [comment]
@loumalouomega I think that now that #12843 is merged this is ready to go as well.

**Comment by loumalouomega** [comment]
> @loumalouomega I think that now that #12843 is merged this is ready to go as well.

Oki

**Review by loumalouomega — COMMENTED** [review]
LGTM

**Comment by rubenzorrilla** [comment]
> > LGTM
> 
> I can approve it if you consider me worthy of that

I think so (you know this class and you're also using it in your side). Note that I reverted the undesired changes from the other branch.

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**